src/backend/api/Fusion.Resources.Domain/Queries/GetRelevantOrgUnits.cs

```diff
@@ -0,0 +1,224 @@
+                {
+
+                    if (org?.FullDepartment != null && org?.Reason != null)
+                    {
+                        var alreadyInList = endResult.FirstOrDefault(x => x.FullDepartment == org.FullDepartment);
+
+                        if (alreadyInList is null)
+                        {
+                            QueryRelevantOrgUnit? data = new QueryRelevantOrgUnit();
+                            if (org.IsWildCard == true)
+                            {
+                                data = cachedOrgUnits.FirstOrDefault(x => x.FullDepartment == org.FullDepartment.Replace('*', ' ').TrimEnd());
+                            }
+                            else
+                            {
+                                data = cachedOrgUnits.FirstOrDefault(x => x.FullDepartment == org.FullDepartment);
+                            }
+
+                            if (data != null)
+                            {
+                                data.Reasons.Add(org.Reason);
+                                endResult.Add(data);
+                            }
+                        }
+                        else
+                        {
+                            if (!alreadyInList.Reasons.Contains(org.Reason))
+                            {
+                                alreadyInList.Reasons.Add(org.Reason);
+                            }
+                        }
+                    }
+                }
+                return endResult;
+            }
+
+            private static List<QueryRelevantOrgUnit> ApplyOdataFilters(ODataQueryParams filter, IEnumerable<QueryRelevantOrgUnit> orgUnits)
+            {
+
+                var filterGenerator = filter.GenerateFilters<QueryRelevantOrgUnit>(m =>
+                {
+                    m.SqlQueryMode = false;
+                    m.MapField("sapId", e => e.SapId);
+                    m.MapField("name", e => e.Name);
+                    m.MapField("shortName", e => e.ShortName);
+                    m.MapField("department", e => e.Department);
+                    m.MapField("fullDepartment", e => e.FullDepartment);
+                    m.MapField("reason", e => e.Reasons);
+                });
+                return orgUnits.Where(filterGenerator.FilterLambda.Compile()).ToList();
+            }
+
+            private async Task<QueryRelatedDepartments?> ResolveCache(string fullDepartmentName, CancellationToken cancellationToken)
+            {
+                return await memCache.GetOrCreateAsync(fullDepartmentName.Trim(), async (entry) =>
+                {
+                    entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(60);
+
+                    var orgUnitResponse = await mediator.Send(new GetRelatedDepartments(fullDepartmentName), cancellationToken);
+                    return orgUnitResponse;
+                });
+            }
+        }
+    }
+
+    internal static class orgUnitAccessReasons
```

Comment:
Capitalization doesn't follow project standard.
Should conform to existing code base. (camel case vs pascal case)